src/types/Typechecker/Errorprinter.hs

```diff
@@ -0,0 +1,244 @@
+
+module Typechecker.Errorprinter (printError, printWarning, noExplanation) where
+
+
+-- Library dependencies
+import Text.PrettyPrint.Annotated
+import Text.PrettyPrint.Annotated.HughesPJ (renderDecoratedM)
+import System.Console.ANSI
+import Text.Printf (printf)
+import Data.Ix(range)
+import Data.Map.Strict (keys)
+
+-- Module dependencies
+import AST.Meta(Position, getPositionFile, getPositions)
+import Identifiers
+import Types
+import Typechecker.Environment
+import Typechecker.TypeError
+import Typechecker.Util
+import Typechecker.Suggestable
+import Typechecker.ExplainTable
+import System.IO
+
+
+
+printError :: TCError -> IO ()
+printError err@(TCError _ Env{bt = []}) =
```

Comment:
the `TCError` ADT was updated from `TCError Error Backtrace` to `TCError Error Env`. However, I believe that you never make use of the environment when you try to print an error. I was wondering if you really need the environment in the `TCError`. I am sure I am wrong, but I could not find where do you make use of the `Env` from the `TCError`.

Reply:
The `Env` is passed along to the long and short suggest. Otherwise we could not evaluate an error and give suggestions.

Reply:
right! I just saw that you need the environment for a specific case of the long suggest. The short suggest doesn't seem to use it

Reply:
short suggest uses it as well to figure out if the name of an unknown method is available as a function. (As I would like to refactor this code into a single function returning both the possible short and long suggestions it would be wise to leave the arguments the same)